code/src/test/kotlin/com/expediagroup/sdk/core/http/RequestTest.kt

```diff
@@ -0,0 +1,286 @@
+    }
+
+    @Test
+    fun `should set one single header`() {
+        // Given
+        val method = Method.GET
+        val url = "https://example.com"
+        val headerName = "content-type"
+        val headerValue = "application/json"
+
+        // When
+        val request = Request
+            .builder()
+            .method(method)
+            .url(url)
+            .setHeader(headerName, headerValue)
+            .build()
+
+        // Expect
+        assertEquals(headerValue, request.headers.get(headerName))
+    }
+
+    @Test
+    fun `should set multiple values for one header`() {
+        // Given
+        val method = Method.GET
+        val url = "https://example.com"
+        val headerName = "content-type"
+        val headerValue1 = "application/json"
+        val headerValue2 = "text/plain"
+
+        // When
+        val request = Request
+            .builder()
+            .method(method)
+            .url(url)
+            .setHeader(headerName, headerValue1)
+            .setHeader(headerName, headerValue2)
+            .build()
+
+        // When
+        assertEquals(listOf(headerValue2), request.headers.values(headerName))
+    }
+
+    @Test
+    fun `should add multiple values for one header as a list`() {
+        // Given
+        val method = Method.GET
+        val url = "https://example.com"
+        val headerName = "content-type"
+        val headerValue1 = "application/json"
+        val headerValue2 = "text/plain"
+
+        // When
+        val request = Request
+            .builder()
+            .method(method)
+            .url(url)
+            .addHeader(headerName, listOf(headerValue1, headerValue2))
+            .build()
+
+        // Expect
+        assertEquals(listOf(headerValue1, headerValue2), request.headers.values(headerName))
+    }
+
+    @Test
+    fun `should set multiple values for one header as a list`() {
+        // Given
+        val method = Method.GET
+        val url = "https://example.com"
+        val headerName = "content-type"
+        val headerValue1 = "application/json"
+        val headerValue2 = "text/plain"
+
+        // When
+        val request = Request
+            .builder()
+            .method(method)
+            .url(url)
+            .setHeader(headerName, listOf(headerValue1, headerValue2))
+            .setHeader(headerName, listOf(headerValue2))
+            .build()
+
+        // Expect
+        assertEquals(listOf(headerValue2), request.headers.values(headerName))
+    }
+
+    @Test
+    fun `should handle adding and setting headers as expected`() {
+        // Given
+        val request = Request
+            .builder()
+            .method(Method.GET)
+            .url("https://example.com")
+            .addHeader("Header1", "Value1")
+            .setHeader("Header2", "Value2")
+            .setHeader("Header1", "NewValue1")
+            .addHeader("Header3", listOf("Value1", "Value2"))
+            .build()
+
+        // When & Expect
```

Comment:
I think the `when` part is merged with `given` above, not with `expect`, Same for other test cases.

Reply:
Done! https://github.com/ExpediaGroup/lodging-connectivity-java-sdk/pull/118/commits/34b3ad93ec49e1f5c370c83167561e0fb1592856